lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,77 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
+            // MariaDB does not follow one rule for versions ;)
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
 
-        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            // Will catch things like '10.2.8-MariaDB-1~lenny-log'
+            preg_match('/(?P<major>\d{1,2})\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $firstMatch);
+            // Will catch things like '5.5.5-10.2.8-MariaDB-10.2.8+maria~xenial-log'
+            preg_match('/\-(?P<major>10)\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $secondMatch);
+
+            if (empty($firstMatch) && empty($secondMatch)
+            ) {
+                throw DBALException::invalidPlatformVersionSpecified(
+                    $version,
+                    'MariaDB-<major_version>.<minor_version>.<patch_version>' .
+                                 'or <major_version>.<minor_version>.<patch_version>-MariaDB'
```

Comment:
How come the `'MariaDB'` marker is part of the version string requirement in the error message but not in the regexp?

Reply:
Good catch, anyway I want to move those tests somewhere else (and at the same time support mysqli detection, see todo list)